### ISSUE TITLE
Report on accuracy

### ISSUE THREAD

**Issue by Sorkanius** [livepeer/verification-classifier#48]
The purpose of this report is to show an insight of what attacks are we detecting and the accuracy on each attack to further discuss a tamper detection model and a quality measure.

**Comment by Sorkanius** [comment]
As promised yesterday we have elaborated this [notebook](https://github.com/livepeer/verification-classifier/blob/master/machine_learning/notebooks/accuracy_report.ipynb) in order to show an insight of the calssification results that we have on the current dataset.

We are only using six features:
1. Dimension
2. Size
3. Mean value of temporal DCT
4. Mean value of temporal gaussian
5. Mean value of temporal gaussian difference
6. Mean value of temporal gaussian threshold

For this report we are focused on the attacks we have simulated that refer to tampering the video. On the examples we have each attack (on 1080p) of this original [video](https://storage.cloud.google.com/livepeer-verifier-originals/0k1ueD_Obac.mp4).
  * [Vignette](https://storage.cloud.google.com/livepeer-verifier-renditions/1080p_vignette/0k1ueD_Obac.mp4).
  * [Vertical flip](https://storage.cloud.google.com/livepeer-verifier-renditions/1080p_flip_vertical/-8ygLPzgpsg.mp4). (A different [source](https://storage.cloud.google.com/livepeer-verifier-originals/-8ygLPzgpsg.mp4) video is shown since original's vertical flip is not available)
  * [Watermark](https://storage.cloud.google.com/livepeer-verifier-renditions/1080p_watermark/0k1ueD_Obac.mp4)
  * [Watermark 345x114](https://storage.cloud.google.com/livepeer-verifier-renditions/1080p_watermark-345x114/0k1ueD_Obac.mp4).
  * [Watermark 856x856](https://storage.cloud.google.com/livepeer-verifier-renditions/1080p_watermark-856x856/0k1ueD_Obac.mp4).
  * [Black and white](https://storage.cloud.google.com/livepeer-verifier-renditions/1080p_black_and_white/0k1ueD_Obac.mp4) (grayscaling the original video)

The dataset contains a total of 134296 videos  and the resolutions used are: 144p, 240p, 360p, 480p, 720p and 1080p. It is split into 3 subsets.
* Training set. Only contains legit assets. The model is trained with **only** these videos. There is a total of **18014** assets.
* Testing set. Only contains legit assets. It is used to test the model with the possitive class. There is a total of **4504** assets.
* Attack set. Only contains assets with the previously mentioned attacks. It is used to test the model with the negative class (which is not present in training). There is a total of **111778** assets.

The global results in classification are:
* TNR: 0.8770
* TPR (train set): 0.9845
* TPR (test set): 0.9860
* F20: 0.9771

The mean distances to the decision function for each set are.
* Train set: 0.6966
* Test set: 0.6865
* Attack set: -4.1140

The test and train set behave similarly. The attack set has a mean negative distance, this means that most of them are classified as attacks (remember the 0.8770 TNR).

Next we zoom in into the test set and see the mean distance and accuracy per resolution:

| Test set | Mean distance | Accuracy |
|:--------:|:-------------:|:--------:|
|   144p   |     0.3851    |  0.9680  |
|   240p   |     0.6458    |  0.9907  |
|   360p   |     0.8575    |  0.9893  |
|   480p   |     1.1545    |  0.9947  |
|   720p   |     0.9644    |  0.9947  |
|   1080p  |     0.1103    |  0.9693  |

As we increase the resolution the distance to the boundary increases until we reach 720p and the distance starts decaying. This a reason why we have to be careful when interpreting the distance to the decision function as  a quality measure. Higher mean distance does not necessarily higher accuracy.

Zooming into the attack set we can see the mean distance and accuracy per resolution: 
| Attack set | Mean distance | Accuracy |
|:----------:|:-------------:|:--------:|
|    144p    |    -3.0060    |  0.7896  |
|    240p    |    -3.4882    |  0.8009  |
|    360p    |    -3.9518    |  0.84593 |
|    480p    |    -4.2606    |  0.8864  |
|    720p    |    -4.7409    |  0.9447  |
|    1080p   |    -5.2364    |  0.9946  |
The attacks from higher resolutions are further from the boundary and are being better classified than at low resolutions.

Now we zoom into which attacks are better detected:

|    Attack type    | Mean distance | Accuracy |
|:-----------------:|:-------------:|:--------:|
|      Vignette     |    -5.4993    |  0.9993  |
|   Vertical flip   |    -5.5074    |  1.0000  |
|     Watermark     |    -3.6916    |  0.8813  |
| Watermark 345x114 |    -1.2098    |  0.5209  |
| Watermark 856x856 |    -4.7276    |  0.9893  |
|  Black and white  |    -5.3089    |  0.9810  |

The vertical flip and attacks such as a different video are detected with no trouble. As the watermark decreases in size it is harder to detect.
